### PR TITLE
small refactor, fix param type

### PR DESCRIPTION
**Pull request by apiraino**
Please, could you also have a look at this pr and let me have your comments?

this pr is somehow connected to https://github.com/FrodeSolheim/fs-uae-launcher/pull/35

See comments in commit

### CODE REVIEW

File: fsgs/Database.py [hunk 1]
```diff
@@ -439,18 +439,11 @@ def get_variant_for_list_and_game(self, list_uuid, game_uuid):
             return None
         return row[0]
 
-    def find_games_new(self, search="", have=3, list_uuid="",
+    def find_games_new(self, search="", have=3, list_uuid=None,
```
Comment (apiraino) [inline fsgs/Database.py]: Since we are checking for `list_uuid` "Noneness", better to have None as default

File: fsgs/Database.py [hunk 2]
```diff
@@ -439,18 +439,11 @@ def get_variant_for_list_and_game(self, list_uuid, game_uuid):
...
-            database_only = True
-            list_uuid = ""
-        elif list_uuid == self.GAME_LIST_CONFIGS:
-            non_database_only = True
-            list_uuid = ""
-        elif list_uuid:
-            have = 0
+
+        # FIXME: game.game_list_game never populated?
+        ENABLE_JOIN = False
```
Comment (apiraino) [inline fsgs/Database.py]: I didn't understand whether we are using table `game.game_list_game` or not.

I have it always empty in my fs-uae-launcher installation, so I'm not sure this JOIN is needed or not: could you clarify this?

File: fsgs/Database.py [hunk 3]
```diff
@@ -538,15 +531,17 @@ def callback(m):
...
-        if database_only:
-            query += " AND path is NULL"
-        elif non_database_only:
-            query += " AND path is NOT NULL"
+        path = ''
+        if list_uuid == self.GAME_LIST_GAMES:
+            path = " AND path IS NULL "
+        elif list_uuid == self.GAME_LIST_CONFIGS:
+            path = " AND path IS NOT NULL "
+        query += path
```
Comment (apiraino) [inline fsgs/Database.py]: here I have simplified a bit the code but it does the same thing.

**Comment by FrodeSolheim** [comment]
Hi, pull request has a couple of problems:
- I don't think your refactoring takes into consideration that "database_only" is also a parameter to the function, and the changes introduce a bug related to this.
- The game list functionality are actually used - but not by many yet

Short explanation: The system supports downloading game lists from oagd.net, you can for example have a list of favorite games, or a list of games you want to play, etc. There is just not yet any user interface for it in FS-UAE launcher.

So that's what the game list selector is used for. These (real) game lists have actual UUIDs. As you can see, I have "hacked together" a rather quick solution where some predefined filters have hardcoded UUIDs.

But since the current code works (even though it may not be pretty), and this pull request may cause problems, and does not really fix anything, I am going to reject it. I appreciate the good intentions though :)

**Pull request closed without merging**